gunicorn/workers/geventlet.py

```diff
@@ -19,22 +17,49 @@
 
 from eventlet import hubs, greenthread
 from eventlet.greenio import GreenSocket
-from eventlet.hubs import trampoline
 from eventlet.wsgi import ALREADY_HANDLED as EVENTLET_ALREADY_HANDLED
 import greenlet
 
 from gunicorn.workers.base_async import AsyncWorker
 
 
-def _eventlet_sendfile(fdout, fdin, offset, nbytes, _os_sendfile=os.sendfile):
-    while True:
-        try:
-            return _os_sendfile(fdout, fdin, offset, nbytes)
-        except OSError as e:
-            if e.args[0] == errno.EAGAIN:
-                trampoline(fdout, write=True)
-            else:
-                raise
+def _eventlet_socket_sendfile(self, file, offset=0, count=None):
+    # Based on the implementation in gevent which in turn is slightly
+    # modified from the standard library implementation.
+    if self.gettimeout() == 0:
+        raise ValueError("non-blocking sockets are not supported")
+    if offset:
+        file.seek(offset)
```

Comment:
Do we have to worry about this not being implemented, or do we want this to raise in the event we try to send from an offset of something unseekable? Alternatively, maybe the argument is here for completeness but is never triggered?

Reply:
Right, that's just matching the `socket.sendfile` signature. gunicorn will never use it.

In the event that some application software calls it on an eventlet socket, it needs to match  (prior to this, eventlet would copy `socket.sendfile` to itself and call that directly, so we need to maintain the signature and semantics as much as possible; of course, it wouldn't actually *work* because the socket was non-blocking, but now that it does and the `ValueError` isn't raised the rest of it should raise the standard exceptions or work the standard way.)